definitions/marts/looker_studio_marts/ecf/ls_ecf_2021_declarations_in_milestone_period.sqlx

```diff
@@ -0,0 +1,188 @@
+    description: "This mart holds all ECF declarations for participants in the 2021 cohort with multiple in any one Milestone Period. \n \
+    Objective: \n \
+      To identify all declarations that fall within the same milestone period for review. \n \
+    Dependencies: \n \
+      static_tables.ecf_2021_milestone_periods \n \
+      dataform.ls_declarations_provider_names \n \
+      dataform.participant_profiles_latest_cpd \n \
+      dataform.schedules_latest_cpd \n \ ",
+    columns: {
+        user_id: "This comes from the teacher profile associated with the participant profile. User ID also exists independently in other tables.",
+        participant_profile_id: "The participant profile ID as per the ECF Induction record.",
+        role: "The role of the participant which is either ECT or Mentor as per the participant profile type.",
+        current_schedule: "The schedule as per the latest participant profile.",
+        cohort: "The cohort/academic year corresponding to when the participant started their course.",
+        declaration_id: "The unique identifier for the declaration.",
+        declaration_type: "The type of declaration which can be started, retained-#, extended-# or completed. #'s range from 1 to 4.",
+        declaration_date: "The date of when the declaration was created.",
+        declaration_state: "The state of the declaration which for this mart can only be either paid, payable or eligible.",
+        declaration_type_sort_order: "A numeric value associated with the declaration type to enable sorting based upon its value. The order is started, retained-[1-4], extended-[1-3], completed.",
+        lead_provider: "The name of the Lead Provider that submitted the declaration.",
+        milestone_period_id: "The integer ID for the Milestone Period as per the static table ecf_2021_milestone_periods.",
+        milestone_period: "Concatenation of period start and end dates to create a grouping. These periods are only relevant to the 2021 academic cohort.",
+        milestone_period_dupe_flag: "1 or 0 if the declaration exists within a Milestone Period alongside another.",
+        declaration_date_dupe_flag: "1 or 0 if the declaration exists on the same Declaration Date as another.",
+        same_milestone_period: "This field contains 1 of 3 values. 'Same milestone period used more than once' if the milestone_period_dupe_flag is 1 and declaration_date_dupe_flag is 0, 'Same declaration date used more than once' if declaration_date_dupe_flag is 1 or blank if neither dupe flag is 1.",
+        is_dummy_date: "TRUE/FALSE if the declaration date is the same as the start or end date of the Milestone Period. Some Lead Providers were informed to use these dates in order to allow multiple declarations in a single period and this flag allows the Contract Management team to assess those declarations.",
+        declaration_date_chk: "Count of declarations with the same date by user_id, provider_name and course_identifier",

```

Comment:
is this user_id or participant_profile_id?

Reply:
because they are not the same

Reply:
This still doesn't match the code choices below where it seems your grouping by participant_profile_id